parcels/plotting.py

```diff
@@ -0,0 +1,272 @@
+                if isinstance(particles.fieldset.U.grid, CurvilinearGrid):
+                    ax.set_xlim(particles.fieldset.U.grid.lon[latS, lonW], particles.fieldset.U.grid.lon[latN, lonE])
+                    ax.set_ylim(particles.fieldset.U.grid.lat[latS, lonW], particles.fieldset.U.grid.lat[latN, lonE])
+                else:
+                    ax.set_xlim(particles.fieldset.U.grid.lon[lonW], particles.fieldset.U.grid.lon[lonE])
+                    ax.set_ylim(particles.fieldset.U.grid.lat[latS], particles.fieldset.U.grid.lat[latN])
+            else:
+                ax.set_xlim(np.nanmin(particles.fieldset.U.grid.lon), np.nanmax(particles.fieldset.U.grid.lon))
+                ax.set_ylim(np.nanmin(particles.fieldset.U.grid.lat), np.nanmax(particles.fieldset.U.grid.lat))
+        elif domain is not None:
+            if isinstance(particles.fieldset.U.grid, CurvilinearGrid):
+                ax.set_extent([particles.fieldset.U.grid.lon[latS, lonW], particles.fieldset.U.grid.lon[latN, lonE],
+                               particles.fieldset.U.grid.lat[latS, lonW], particles.fieldset.U.grid.lat[latN, lonE]])
+            else:
+                ax.set_extent([particles.fieldset.U.grid.lon[lonW], particles.fieldset.U.grid.lon[lonE],
+                               particles.fieldset.U.grid.lat[latS], particles.fieldset.U.grid.lat[latN]])
+
+    else:
+        if field is 'vector':
+            field = particles.fieldset.UV
+        elif not isinstance(field, Field):
+            field = getattr(particles.fieldset, field)
+
+        plt, fig, ax, cartopy = plotfield(field=field, animation=animation, show_time=show_time, domain=domain,
+                                          projection=projection, land=land, vmin=vmin, vmax=vmax, savefile=None,
+                                          titlestr='Particles and ')
+        if plt is None:
+            return  # creating axes was not possible
+
+    if with_particles:
+        plon = np.array([p.lon for p in particles])
+        plat = np.array([p.lat for p in particles])
+        if cartopy:
+            ax.scatter(plon, plat, s=20, color='black', zorder=20, transform=cartopy.crs.PlateCarree())
+        else:
+            ax.scatter(plon, plat, s=20, color='black', zorder=20)
+
+    if animation:
+        plt.draw()
+        plt.pause(0.0001)
+    elif savefile is None:
+        plt.show()
+    else:
+        plt.savefig(savefile)
+        logger.info('Plot saved to ' + savefile + '.png')
+        plt.close()
+
+
+def plotfield(field, show_time=None, domain=None, projection=None, land=None,
+              vmin=None, vmax=None, savefile=None, **kwargs):
+    """Function to plot a Parcels Field
+
+    :param show_time: Time at which to show the Field
+    :param domain: Four-vector (latN, latS, lonE, lonW) defining domain to show
+    :param projection: type of cartopy projection to use (default PlateCarree)
+    :param land: Boolean whether to show land
```

Comment:
Here (and everywhere else where there is a `land` flag).
The default value for `land` is `None`, while it is a boolean.
I guess it is because you want `land=True` if there is a geomap, and `False` otherwise?
I would still put it to `True` by default, since it is clear, and simply put in the docstrings that this value is ignored for flat mesh, no?